### PR TITLE
upgraded pytest version to 8

### LINKED ISSUE #187: Bump pytest to version 8
https://github.com/PolicyEngine/policyengine-core/blob/4c67661f0e7e23d526f5bc99af91f8b54a9bd0c9/setup.py#L13

Now on 8.1.1: https://pypi.org/project/pytest/

### PR DESCRIPTION
**Pull request by masterismail**
fixes #187

### CODE REVIEW

**Comment by anth-volk** [comment]
@masterismail I'm sorry, I didn't even realize this PR was open, as we don't monitor this repo's PRs as closely. Would you mind adding a changelog entry?

**Comment by masterismail** [comment]
No problem at all @anth-volk, I too did't realise that a test was failing. Now added a changelog entry.

**Comment by anth-volk** [comment]
Weirdly, these tests fail, claiming that the version you input already exists. Would you mind rebasing and pushing that? Sorry for the hassle.

**Comment by anth-volk** [comment]
We're having the same issue. To make sure, did you fully pull the live repo into your fork's `master`, then rebase off of that? I think that may be the issue here.

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    added:
+    - upgraded the pytest version to 8
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - Dependency upgrade for pytest, from 7 to 8.

```

**Comment by masterismail** [comment]
> We're having the same issue. To make sure, did you fully pull the live repo into your fork's `master`, then rebase off of that? I think that may be the issue here.

I had run `.github/is-version-number-acceptable.sh` guess that's causing the issue

**Comment by masterismail** [comment]
sorry for inconvinence, fixed this issue here #204

File: changelog_entry.yaml (hunk 1 above)
Comment (masterismail) [reply changelog_entry.yaml:4]: closing in the light of #187

**Pull request closed without merging**